hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/precompile/codec/CustomFee.java

```diff
@@ -44,4 +46,57 @@ public FractionalFee getFractionalFee() {
     public RoyaltyFee getRoyaltyFee() {
         return royaltyFee;
     }
+
+    public boolean hasFixedFee() {
+        return fixedFee != null;
+    }
+
+    public boolean hasFractionalFee() {
+        return fractionalFee != null;
+    }
+
+    public boolean hasRoyaltyFee() {
+        return royaltyFee != null;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(fixedFee, fractionalFee, royaltyFee);
+    }
+
+    @SuppressWarnings("java:S1126")
```

Comment:
Re-write this using `java.util.Objects.equals(Object, Object)`.  It's null aware and this can be reduced to a single chained return statement, satisfying the sonar warning.  Going through methods is entirely unneeded as this method is access it's own fields, encapsulation is not broken.

Something like this - https://github.com/hashgraph/hedera-services/blob/82c2d4e20642f29568284f1cee763b08f4f01959/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/legacy/core/jproto/TxnReceipt.java#L238

I expect it is what the IntelliJ auto-generated equals would generate.